python/geom.py

```diff
@@ -615,7 +628,10 @@ def update_weights(self,x):
         """
         if x.size != self.num_params:
             raise ValueError("weights of shape {} do not match user specified grid dimension: {}".format(self.weights.size,self.grid_size))
-        self.weights[:]=x.flatten().astype(np.float64)
+        if np.amin(x) < 0. or np.amax(x) > 1.:
```

Comment:
Rather than duplicating the code used for the check, you should create a method

Reply:
Done.